packages/react-data-grid/src/Canvas.tsx

```diff
@@ -171,35 +177,64 @@ export default function Canvas<R>({
     const rows = [];
 
     for (let idx = rowOverscanStartIdx; idx <= rowOverscanEndIdx; idx++) {
-      rows.push(renderRow(idx));
+      const row = rowGetter(idx);
+      rows.push(renderRow(row, idx));
     }
 
     return rows;
   }
 
-  function renderRow(idx: number) {
-    const row = rowGetter(idx);
-    const rendererProps: RendererProps<R> = {
+  function getBaseRendererProps(row: R, idx: number): Omit<RendererProps<R>, 'ref'> {
+    const { columns, lastFrozenColumnIndex } = columnMetrics;
+    return {
       key: idx,
-      ref(row) {
-        if (row) {
-          rows.set(idx, row);
-        } else {
-          rows.delete(idx);
-        }
-      },
       idx,
       row,
       height: rowHeight,
-      columns: columnMetrics.columns,
-      isRowSelected: isRowSelected(row),
+      columns,
+      isRowSelected: false,
       onRowSelectionChange,
       cellMetaData,
-      subRowDetails: getSubRowDetails ? getSubRowDetails(row) : undefined,
       colOverscanStartIdx,
       colOverscanEndIdx,
-      lastFrozenColumnIndex: columnMetrics.lastFrozenColumnIndex,
-      scrollLeft
+      lastFrozenColumnIndex,
+      scrollLeft,
+      isSummaryRow: false
+    };
+  }
+
+  function renderSummaryRow(row: R, idx: number) {
```

Comment:
This component is becoming little complex to understand, I will check if we can cleanup somehow

Reply:
https://github.com/adazzle/react-data-grid/pull/1787 should address it